tests/transforms/test_decompositions.py

```diff
@@ -453,153 +482,115 @@ def test_zxz_decomposition_jax(self, U, expected_gates, expected_params):
         config.update("jax_enable_x64", True)
 
         U = jax.numpy.array(U, dtype=jax.numpy.complex128)
-
         obtained_gates = one_qubit_decomposition(U, Wires("a"), "ZXZ", return_global_phase=True)
+        _run_assertions(U, expected_gates, expected_params, obtained_gates)
 
-        self._run_assertions(U, expected_gates, expected_params, obtained_gates)
+        # Restore the configuration
+        config.update("jax_enable_x64", remember)
 
 
-single_qubit_decomps = [
-    (
-        np.array(
-            [
-                [0, -9.831019270939975e-01 + 0.1830590094588862j],
-                [9.831019270939975e-01 + 0.1830590094588862j, 0],
-            ]
-        ),
-        "ZYZ",
-        typeof_gates_zyz,
-        [12.382273469673908, np.pi, 0.18409714468526372, 0],
-    ),
-    (
-        np.array(
-            [
-                [-0.28829348 - 0.78829734j, 0.30364367 + 0.45085995j],
-                [0.53396245 - 0.10177564j, 0.76279558 - 0.35024096j],
-            ]
-        ),
-        "XYX",
-        typeof_gates_xyx,
-        (
-            10.845351366405708,
-            1.3974974118006183,
-            0.45246583660683803,
-            1.1759220332464762,
-        ),
-    ),
-    (
-        np.array(
-            [
-                [0, -9.831019270939975e-01 + 0.1830590094588862j],
-                [9.831019270939975e-01 + 0.1830590094588862j, 0],
-            ]
-        ),
-        "ZXZ",
-        typeof_gates_zxz,
-        [10.811477142879012, np.pi, 1.7548934714801607, 0],
-    ),
-]
-
-
-rot_decomps = [
+single_qubit_decomps_rot = [
     # These will be decomposed to RZ
-    (I, "rot", qml.RZ, [0.0]),
-    (Z, "rot", qml.RZ, [np.pi]),
-    (S, "rot", qml.RZ, [np.pi / 2]),
-    (T, "rot", qml.RZ, [np.pi / 4]),
-    (qml.RZ(0.3, wires=0).matrix(), "rot", qml.RZ, [0.3]),
-    (qml.RZ(-0.5, wires=0).matrix(), "rot", qml.RZ, [-0.5]),
+    (I, qml.RZ, [0.0]),
+    (Z, qml.RZ, [np.pi]),
+    (S, qml.RZ, [np.pi / 2]),
+    (T, qml.RZ, [np.pi / 4]),
+    (qml.RZ(0.3, wires=0).matrix(), qml.RZ, [0.3]),
+    (qml.RZ(-0.5, wires=0).matrix(), qml.RZ, [-0.5]),
     # This will be decomposed to Rot
     (
         qml.Rot(0.2, 0.5, -0.3, wires=0).matrix(),
-        "rot",
         qml.Rot,
         [0.2, 0.5, -0.3],
     ),
 ]
 
 
-class TestOneQubitUnitaryDecomposition:
-    """Test that one qubit unitaries are correctly decomposed."""
+def _run_assertions_rot(U, expected_gate, expected_params, obtained_gates):
```

Comment:
Why was it impossible to use the other `_run_assertions`?

Reply:
It's because the rotation decomposition behaves differently from the other ones. The other ```_run_assertions``` is written exclusively for decompositions into 4 gates, specifically three rotations and a global phase, whereas the rotation decomposition returns just one gate. I mean they could use the same ```_run_assertions``` method if we do a bit of refactoring, but the other ```_run_assertions``` is not readily available to be used on the rot decomposition.

Reply:
After some examination, I feel that the single qubit decompositions module and the unit tests need some cleaning up that is beyond the scope of this PR. This PR is meant to add a new type of decomposition without drastically changing other parts of the code. I have opened up a new issue (https://github.com/PennyLaneAI/pennylane/issues/4868) addressing some of the inconsistencies in the module, and I plan to open another PR after this one is merged, where I will rewrite the assertion logic to work on all types of single qubit decompositions.

Reply:
Update: I have submitted another pull request that cleans up the tests further. https://github.com/PennyLaneAI/pennylane/pull/4869

Reply:
Thank you for creating a second PR, I will take a look this morning.

Reply:
@rmoyard Thank you. The second PR builds on top of this one, so right now it contains all changes in this PR as well. It might be easier to identify the changes exclusive to the second PR after this one is merged.